src/python_testing/TC_CADMIN_1_15.py

```diff
@@ -0,0 +1,244 @@
+        self.th3 = th3_fabric_admin.NewController(nodeId=3, useTestCommissioner=True)
+
+        self.step(2)
+        GC_cluster = Clusters.GeneralCommissioning
+        attribute = GC_cluster.Attributes.BasicCommissioningInfo
+        duration = await self.read_single_attribute_check_success(endpoint=0, cluster=GC_cluster, attribute=attribute)
+        self.max_window_duration = duration.maxCumulativeFailsafeSeconds
+
+        self.step(3)
+        fabrics = await self.get_fabrics(th=self.th1)
+        initial_number_of_fabrics = len(fabrics)
+
+        self.step(4)
+        params = await self.OpenCommissioningWindow(th=self.th1, expectedErrCode=0x00)
+        setupPinCode = params.setupPinCode
+
+        self.step(5)
+        await self.CommissionAttempt(setupPinCode, thnum=2, th=self.th2)
+
+        self.step(6)
+        params2 = await self.OpenCommissioningWindow(th=self.th1, expectedErrCode=0x00)
+        setupPinCode2 = params2.setupPinCode
+
+        self.step(7)
+        await self.CommissionAttempt(setupPinCode2, thnum=3, th=self.th3)
+
+        self.step(8)
+        fabrics = await self.get_fabrics(th=self.th2)
+        if len(fabrics) != initial_number_of_fabrics + 2:
+            # len of fabrics is expected to be 3, if 3 not found then we assert failure here
+            asserts.fail("Expected number of fabrics not correct")
+
+        self.step(9)
+        # TODO: Currently on hold for impl created by Raul for DNS-SD check of multiple operational service records
+
+        self.step(10)
+        fabric_idx_cr2 = await self.read_currentfabricindex(th=self.th2)
+
+        self.step(11)
+        removeFabricCmd = Clusters.OperationalCredentials.Commands.RemoveFabric(fabric_idx_cr2)
+        await self.th2.SendCommand(nodeid=self.dut_node_id, endpoint=0, payload=removeFabricCmd)
+
+        self.step(12)
+        # Verifies TH_CR2 is unable to read the Basic Information Cluster’s NodeLabel attribute of DUT_CE as no longer on network
+        try:
+            await self.read_single_attribute_check_success(
+                dev_ctrl=self.th2,
+                endpoint=0,
+                cluster=Clusters.BasicInformation,
+                attribute=Clusters.BasicInformation.Attributes.NodeLabel
+            )
+            asserts.fail("Expected exception not thrown")
+        except ChipStackError as e:
+            # Verify that the DUT returns an "Timeout" status response
+            asserts.assert_equal(e.err, 0x00000032,
+                                 "Expected to timeout as DUT_CE is no longer on network")
+
+        self.step(13)
+        fabrics2 = await self.get_fabrics(th=self.th1)
+        fabric_indexes = [fabric.fabricIndex for fabric in fabrics2]
+        if len(fabrics2) != initial_number_of_fabrics + 1:
+            # len of fabrics is expected to be 2, if 2 not found then we assert failure
+            asserts.fail(f"Expected number of fabrics not correct, should show 2, but instead shows {str(len(fabrics2))}")
```

Comment:
I don't actually care too much, but you don't need str operator in an f string. less typing.

Reply:
Thank you, I will remember and change this!